src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -1107,7 +1107,7 @@ var ReactCompositeComponent = {
     if (__DEV__) {
       var componentName = component && component.getName ?
         component.getName() : 'a component';
-      warning(publicComponentInstance != null,
```

Comment:
Can we check `this._compositeType` instead, and only warn for stateless functional components? It seems like this would solve the issue by making the warning more specific without hardcoding information about test renderer. Another upside is that the warning would still fire if the to use refs with stateless components incorrectly while using test renderer for example.